### PR TITLE
feat: allow to specify app id when creating a nuxt app

### PR DESCRIPTION
**Pull request by tobiasdiez**
### 📚 Description

When creating a new nuxt application via `createNuxtApp`, it is now possible to overwrite the app id specified in the nuxt config.

This is needed for the storybook module where we create multiple instances of the same nuxt app (in the same website) and would like to give them different ids to preserve their context. See https://github.com/nuxt-modules/storybook/pull/723#issuecomment-2250731275 for more context that triggered this PR.

### CODE REVIEW

**Review by huang-julien — APPROVED** [review]

File: packages/nuxt/src/app/nuxt.ts [hunk 1]
```diff
@@ -244,13 +244,17 @@ export type ObjectPluginInput<Injections extends Record<string, unknown> = Recor
 export interface CreateOptions {
   vueApp: NuxtApp['vueApp']
   ssrContext?: NuxtApp['ssrContext']
+  /**
+   * The name of the Nuxt application, overrides the default name specified in the Nuxt config (default: `nuxt-app`).
+   */
+  appId?: NuxtApp['_name']
```
Comment (danielroe) [inline packages/nuxt/src/app/nuxt.ts:250]: Would it make sense for this to be `name` instead?

Comment (tobiasdiez) [reply packages/nuxt/src/app/nuxt.ts:250]: I was unsure about this as well. In the nuxt config it is appId, but name would be a natural choice as well. Do you have a preference?

Comment (tobiasdiez) [reply packages/nuxt/src/app/nuxt.ts]: I agree, this makes sense! 
However, there was already an `_id` field in the `NuxtApp`, which was used to generate app-wide unique ids in `useId`. I've renamed that field to `_genId`.

**Review by danielroe — APPROVED** [review]

**Comment by tobiasdiez** [comment]
@danielroe @huang-julien I've tried to use this PR in the storybook module but encountered the following error:
```
Error: [nuxt] A composable that requires access to the Nuxt instance was called outside of a plugin, Nuxt hook, Nuxt middleware, or Vue setup function. This is probably not a Nuxt bug. Find out more at `https://nuxt.com/docs/guide/concepts/auto-imports#vue-and-nuxt-composables`.
  at definePayloadReviver (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/nuxt@3.13.0_@parcel+watcher@2.4.1_@types+node@22.5.0_encoding@0.1.13_eslint@9.9.0_jiti@1.21.6_3xmfrk4x4fipugkjlwslnycjnu/node_modules/nuxt/dist/app/composables/payload.js?v=728a24b5:125:5)
  at setup (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/nuxt@3.13.0_@parcel+watcher@2.4.1_@types+node@22.5.0_encoding@0.1.13_eslint@9.9.0_jiti@1.21.6_3xmfrk4x4fipugkjlwslnycjnu/node_modules/nuxt/dist/app/plugins/revive-payload.client.js?v=728a24b5:40:27)
  at /@id/virtual:nuxt:D:/Programming/nuxt-storybook/examples/showcase/.nuxt/plugins/client.mjs:33:69
  at executeAsync (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/unctx@2.3.1/node_modules/unctx/dist/index.mjs?v=728a24b5:111:19)
  at setup (/@id/virtual:nuxt:D:/Programming/nuxt-storybook/examples/showcase/.nuxt/plugins/client.mjs:33:43)
  at /@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/nuxt@3.13.0_@parcel+watcher@2.4.1_@types+node@22.5.0_encoding@0.1.13_eslint@9.9.0_jiti@1.21.6_3xmfrk4x4fipugkjlwslnycjnu/node_modules/nuxt/dist/app/nuxt.js?v=728a24b5:140:66
  at fn (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/nuxt@3.13.0_@parcel+watcher@2.4.1_@types+node@22.5.0_encoding@0.1.13_eslint@9.9.0_jiti@1.21.6_3xmfrk4x4fipugkjlwslnycjnu/node_modules/nuxt/dist/app/nuxt.js?v=728a24b5:217:44)
  at runWithContext (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/@vue+runtime-core@3.4.38/node_modules/@vue/runtime-core/dist/runtime-core.esm-bundler.js?v=728a24b5:2682:18)
  at callWithNuxt (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/nuxt@3.13.0_@parcel+watcher@2.4.1_@types+node@22.5.0_encoding@0.1.13_eslint@9.9.0_jiti@1.21.6_3xmfrk4x4fipugkjlwslnycjnu/node_modules/nuxt/dist/app/nuxt.js?v=728a24b5:223:24)
  at createNuxtApp/runWithContext (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/nuxt@3.13.0_@parcel+watcher@2.4.1_@types+node@22.5.0_encoding@0.1.13_eslint@9.9.0_jiti@1.21.6_3xmfrk4x4fipugkjlwslnycjnu/node_modules/nuxt/dist/app/nuxt.js?v=728a24b5:37:53)
  at run (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/@vue+reactivity@3.4.38/node_modules/@vue/reactivity/dist/reactivity.esm-bundler.js?v=728a24b5:32:16)
  at runWithContext (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/nuxt@3.13.0_@parcel+watcher@2.4.1_@types+node@22.5.0_encoding@0.1.13_eslint@9.9.0_jiti@1.21.6_3xmfrk4x4fipugkjlwslnycjnu/node_modules/nuxt/dist/app/nuxt.js?v=728a24b5:37:31)
  at applyPlugin (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/nuxt@3.13.0_@parcel+watcher@2.4.1_@types+node@22.5.0_encoding@0.1.13_eslint@9.9.0_jiti@1.21.6_3xmfrk4x4fipugkjlwslnycjnu/node_modules/nuxt/dist/app/nuxt.js?v=728a24b5:140:39)
  at executePlugin (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/nuxt@3.13.0_@parcel+watcher@2.4.1_@types+node@22.5.0_encoding@0.1.13_eslint@9.9.0_jiti@1.21.6_3xmfrk4x4fipugkjlwslnycjnu/node_modules/nuxt/dist/app/nuxt.js?v=728a24b5:159:34)
  at applyPlugins (/@fs/D:/Programming/nuxt-storybook/node_modules/.pnpm/nuxt@3.13.0_@parcel+watcher@2.4.1_@types+node@22.5.0_encoding@0.1.13_eslint@9.9.0_jiti@1.21.6_3xmfrk4x4fipugkjlwslnycjnu/node_modules/nuxt/dist/app/nuxt.js?v=728a24b5:190:11)
  at /@fs/D:/Programming/nuxt-storybook/packages/storybook-addon/dist/preview.mjs:40:9
```

when calling `applyPlugins` (see https://github.com/nuxt-modules/storybook/pull/762)

The reason is that I do set the context via `getContext("nuxt-app-customid").set(nuxt, true)` but later `definePayloadReviver` calls `useNuxtApp()` without any `appId`, which thus falls back to the id `nuxt-app`; which is not set. As a workaround I now use `getContext('nuxt-app').set(nuxt, true)`. This works but feels a bit like a hack (and I'm not sure if it covers all edge cases).

How is `useNuxtApp()` (without a nuxt-app id as argument) supposed to work in a multi-app context? Moreover, should `callWithNuxt` also set the default `nuxt-app` context to the passed nuxt instance?


(I can open a new issue for these questions, but thought this PR has the required context.)

**Comment by danielroe** [comment]
@tobiasdiez Yes, multi-app support hasn't yet landed. The next step would be a compile time injection of the nuxt app-id in calls to `useNuxtApp`. For now, it's definitely safer to use `nuxt-app` as the app id, and I think your workaround is fine.

If you are able to add some tests we can add it to https://github.com/nuxt/ecosystem-ci/ so we ensure any future changes along these lines don't break the storybook integration. 🙏

**Comment by tobiasdiez** [comment]
> @tobiasdiez Yes, multi-app support hasn't yet landed. The next step would be a compile time injection of the nuxt app-id in calls to `useNuxtApp`. For now, it's definitely safer to use `nuxt-app` as the app id, and I think your workaround is fine.

Thanks for the feedback! I really like the multi-app feature and the possibilities it opens. Big thanks to everyone that worked on it so far!

> If you are able to add some tests we can add it to https://github.com/nuxt/ecosystem-ci/ so we ensure any future changes along these lines don't break the storybook integration. 🙏

Sure! https://github.com/nuxt/ecosystem-ci/pull/14

**Pull request merged**